### ISSUE TITLE
Simplified Chinese contributor's list doesn't match & copyright problems

### ISSUE THREAD

**Issue by TheInterestingSoul** [SeaDve/Kooha#173]
I'd like to make a brief introduction about what happened:
- There are seven people involved Simplified Chinese translation of Kooha (including me), but only four of them are shown on Kooha's about page;
- It's obvious that someone removed another contributor's name without permission;
- @lumingzh added his name to the list without any valid contribution.

# Copyright Problems
In July 2021, after reading an advertising article about Kooha, I made the first version of the Simplified Chinese translation (commit https://github.com/SeaDve/Kooha/commit/2d0ca61b758dd63a70dd12616471ba2bef6f2b90 and https://github.com/SeaDve/Kooha/commit/a0121217319571b4e6a42a7b1a01bb98c213cab2) on Weblate. However, I'm used to using OBS Studio for screen recording, and I've ignored Kooha for about one year.
Today it suddenly backed to my mind, so I checked its translation record on Weblate. And to my surprise, I could not find my name on the credit list.
![Screenshot_20220729_211813](https://user-images.githubusercontent.com/41455159/181768872-08774449-44a2-4de3-a2e1-28cf7863b098.png)
(You can check the history of `zh_Hans.po` on both [GitHub](https://github.com/SeaDve/Kooha/commits/main/po/zh_Hans.po) and [Weblate](https://hosted.weblate.org/translate/kooha/pot-file/zh_Hans/?checksum=deecda9491124d4f#history).)
It seems that [sliencezhou4321\@Weblate](https://hosted.weblate.org/user/silencezhou4321/)(no GitHub account), has deleted my name somehow. I'm not sure whether I could recover my name by myself, so I found his email address on Weblate and start negotiating with him.

# First Contact
I said:  
> Credits直接删掉了我的贡献……删掉了我的贡献……删掉了我的贡献……我又去看后来的记录：除了我的记录被你删掉以外，其他人都不约而同地，把自己的名字加在在别人的后面
>
> (My name has been removed from the list of translators... I went to see the later records: except for my name, which you deleted, everyone else unfailingly put their names after fellow translators.)
>
> 我不知道您是什么情况，看您在weblate的翻译记录也很多的啊，如果说不知道贡献者的基本规则也说不过去……我也不是觉得我翻译了这款软件很有价值，但我第一眼看到的时候，不可不谓震惊……
>
> (I don't know why you remove my name. Since there are a lot of translation records in your Weblate, it is unreasonable for you not to know the basic rules of contribution... I don't think it's so precious that I translated this software, but when I first saw it, I was shocked...)

He replied immediately:

> 抱歉，您看到的这个我也第一次看到，先让我去核实一下到底发生了什么，因为我很久没有去这个网站看过了。 如果真的是我不小心删掉了你的贡献，我会帮忙修复它，非常抱歉。
>
> (Sorry, I saw this for the first time too, but let me check what happened first because I haven't seen Weblate for a long time. If it's me who accidentally deleted your contribution, I'll help fix it, I'm sorry.)
>
> 讲道理，我肯定不会做这样的事，完全没必要啊，但是我一般是直接用这个平台翻译的，如果出现这样的事我该有印象的，但一点都没有。网站还在登录，您稍等。。。。或者可以加我微信\[his Wechat ID, private\]"
>
> (To be frank, I certainly will not do such a thing, it is completely unnecessary. I usually use this platform to translate directly, If such a thing happened, I should have been able to remember it, but it wasn't. I'm attempting to log in to my Weblate account... Or you can ask me on WeChat \[his WeChat ID, private\])

While he was working to find my translation record, I went on inspecting the commit history on GitHub, and I discovered more interesting things.

# More than one person
One can check the latest version of `zh_Hans.po`, its header showed that there have been seven people submitting their translation:

https://github.com/SeaDve/Kooha/blob/95a482ca8e3c40ebdd93e49fdecc198355f0f84f/po/zh_Hans.po#L4-L10

But only four of them were presented on the software about page:

https://github.com/SeaDve/Kooha/blob/95a482ca8e3c40ebdd93e49fdecc198355f0f84f/po/zh_Hans.po#L255-L262

What's more, @lumingzh submitted a translation on Apr 17th with no valuable content but added his name to the translator list. https://github.com/SeaDve/Kooha/commit/8c300b1802941cc4a32c9b9878fcc7f0bc6cbd4a

# Recovering My Name
Finally, sliencezhou4321 checked my record and tried to recover my name. He made a "suggested translation" and said:
> 对不起，我找到你被删掉的签名。。我不明白为什么会发生这样的事情，当时是公司的一个任务，需要在开源软件做翻译工作，然后我刚开始用这个软件可能不太熟悉？没看懂？总之是我的疏忽造成的，非常抱歉。
> 是这样的，我看这个小软件的翻译已经完成了，我先去尝试把这个记录改回来，增加你的签名在前面或者把我的签名改掉，但是由于没有新的翻译更新，因此我不确定这个操作是不是会通过。那如果没通过，我帮你在其它软件上留一些签名可以吗？
非常抱歉尽管我不是故意的但还是造成了您的困扰，希望可以有办法补偿你。"
> 
> ![Catch](https://user-images.githubusercontent.com/41455159/181769387-13f04817-d49d-471f-b186-f46cb603dbe1.jpg)
>
> (Sorry, I found your deleted signature. I don't understand why this happened, it was a task for the company at that time to do translation work in OSS, and then I may not be familiar with Weblate system. Didn't All in all it's caused by my carelessness, very sorry.
> The translation of Kooha has been completed, I first tried to change this record back, adding your name to the top of the list or even replacing my signature, but since there is no other update, I am not sure if this operation will pass. So if it doesn't work, can I help you leave some signatures on other software?
> I'm very sorry that I didn't mean to bother you, and I hope there's a way to compensate you.)

To be fair, I think he may have just been misled by Weblate's "auto-check format" function, and delete the original contributor accidentally; However, @lumingzh's behavior is not a careless mistake.
# My Advice
As for other language communities, I don't know what the situation is. But for more active languages, such as French, the situation can be even worse than in Simplified Chinese.
Although it is very convenient to use Weblate, you will find problems after reading my story above. Weblate doesn't directly link one's contribution to the software, so it's required to fill an extra field `translator-credits`. Someone may forget to fill it, and someone even tries to get awarded without effort.
I suggest that Kooha separate the list of contributors from Weblate, and write it manually by developers. This prevents both missing contributors and unearned gains. Plus, even if developers don't speak a foreign language, they can at least review whether translations are empty.

**Comment by lumingzh** [comment]
Yes, you're right. I have no contributions to kooha, but I have contributions to mousai which is under the same project "FOSS Projects". So it's weblate's problem not mine.
Thank you.

**Comment by TheInterestingSoul** [comment]
> Yes, you're right. I have no contributions to kooha, but I have contributions to mousai which is under the same project "FOSS Projects". So it's weblate's problem not mine. Thank you.

估计是weblate那个语料库的问题，mousai翻译后，相同字符串 `translator-credits` 被同步到了kooha，这就是weblate的坑

**Comment by SeaDve** [comment]
It might be caused by Weblate synchronizing strings across components since Mousai, Kooha, and Breathing (the components) share the same project.

This could be fixed by disabling the sharing between components, but not sure if everyone agrees with that. Personally, I would disable it if that would fix the issue.
